networks/p2p/simulations/adapters/inproc.go

```diff
@@ -202,12 +203,13 @@ func (sn *SimNode) Client() (*rpc.Client, error) {
 
 // ServeRPC serves RPC requests over the given connection by creating an
 // in-memory client to the node's RPC server
-func (sn *SimNode) ServeRPC(conn net.Conn) error {
+func (sn *SimNode) ServeRPC(conn *websocket.Conn) error {
 	handler, err := sn.node.RPCHandler()
 	if err != nil {
 		return err
 	}
-	handler.ServeCodec(rpc.NewJSONCodec(conn), rpc.OptionMethodInvocation|rpc.OptionSubscriptions)
```

Comment:
Don't we need these rpc options anymore?

Reply:
yes. In the previous logic, only OptionSubscription was checked, and this option was turned off only for the http server. instead of codec option, allowSubscribe flag in handler.go will handle this logic.